### DISCUSSION TITLE
Solarpumpenbetrieb

### DISCUSSION THREAD

**Discussion by AlexKusnezov** [the78mole/esphome_components#73] [Q&A] [answered]
Hey, ich habe das Modul seit knapp einem Jahr in Betrieb und bin mehr als zufrieden damit. Alles funktioniert wie es sein soll und macht unglaublich Spaß die Daten auszuwerten und Automationen zu optimieren ;)

Zusätzlich betreibe ich meine Logamatic 2107 mit einer Solarpumpe and die vier Kollektoren angeschlosssen sind.
Als Sensor gibt der K271 bereits ein "Solarpumpe Absenkung" durch, welcher eben die Leistung reduziert, sobald das Kühlmittel von der Sonne nicht überhitzt. Korrigiert mich bitte wenn ich da was falsches erzähle, ist nur eine Vermutung.(Das wäre wenn die Solatpumpe quasi ausgeht)

Allerdings zeigt auch der 2107 immer ein Symbolb an, sobald diese anspringt, also eine Anzeige für den Solarpumpenbetrieb. 

Könnte man das evtl. auch für den K271 nachrüsten?
Das wäre dann ein ähnlicher Sensor wie der für den Ladepumpenbetrieb.

Freue mich über Antwort und Grüße aus Leonberg.

**Comment by the78mole** [comment]
Hi Alex,

ich kann mir nicht ganz vorstellen, wie genau das funktioniert... Aber ich kann Dir eine Übersicht aller Parameter geben, welche die Buderus liefert bzw. die man der Buderus vorgeben kann: https://github.com/the78mole/esphome_components/blob/main/components/km271_wifi/const.py

Aus den Namen müsste man schließen können, was es ist.

Eventuell kannst Du ja mal ein paar Fotos machen, was genau da angezeigt wird oder was Du steuern oder erfassen möchtest...

Grüße,
Daniel

**Comment by AlexKusnezov** [comment]
@the78mole danke, konnte leider nichts in der const.py finden. 
Das mit der Solarpumpe habe ich aus dem Handbuch entnommen, aber dieses Symbol bezieht sich wahrscheinlich auf die Ladepumpe.
![Screenshot_20240412-150922_Drive](https://github.com/the78mole/esphome_components/assets/33514406/2be7e9dc-1146-4136-b900-126545987c1d)
Ich glaube meine Solarpumpe ist eben gar nicht an den 2107 angeschlossen, sondern einfach nur an den Pufferspeicher.
Solarpumpe:
![2024-04-12-15-14-51-816](https://github.com/the78mole/esphome_components/assets/33514406/c88056cf-858a-4ff3-94aa-e4ce34475ec4)
Von da geht es zum Pufferspeicher: 
![2024-04-12-15-15-00-876](https://github.com/the78mole/esphome_components/assets/33514406/def9b7e0-c682-4844-8a7c-0d61188c4359)
An den ebenfalls der 2107 angeschlossen ist:
![2024-04-12-15-15-09-029](https://github.com/the78mole/esphome_components/assets/33514406/aa1ef636-64c0-43e3-8d72-27bbb072c1ad)
Hab allerdings auch noch den hier fürs Holz(R2104):
![2024-04-12-15-21-48-567](https://github.com/the78mole/esphome_components/assets/33514406/5f06719b-b0f7-4674-a7a8-004b252d3a68)
Und den hier, der wahrscheinlich den Verkehr zwischen den beiden Quellen regelt:
![2024-04-12-15-23-25-708](https://github.com/the78mole/esphome_components/assets/33514406/b612053d-04b8-47a3-aa41-0e5826a0a2e8)
Wahrscheinlich kommunizieren die nur über Relais und Fühler ohne irgendwelche Codes zu generieren, aber vll hast du ja eine Idee wie ich mehr Daten raus bekommen könnte?

**Comment by the78mole** [comment, marked as answer]
Hi Alex,

da könntest Du Recht haben mit der "Relais-Kommunikation"... Vermutlich ist die Solarintegration ein einfacher Differenzregler (Temperatur Puffer < Solarthermie => Pumpe an). Es gibt zwar ein Solarreglermodul, das man in die Logamatic einbauen kann, aber dann ist der Platz für die KM271 belegt.

Mehr daten bekommt man eigentlich nur aus den Installationshandbüchern der einzelnen Teile und die sind manchmal nicht so einfach zu finden. Und dann ist natürlich noch die genaue Verdrahtung wichtig. Nachdem ich auch kein Heizungsfachmann bin und mir nur das angeeignet habe, was ich selbst so brauche, fehlt mir da schlicht dr Überblick, was es so alles für Komponenten von Buderus (und anderen) am Markt gibt. Ich habe auch mittlerweile den Eindruck, selbst Sanitärunternehmen kennen nur eine Hand voll Systeme von maximal 3 Herstellern ziemlich gut. Ich persönlich kenne halt meine Logamatic (mit Brenner) und die Panasonic-Monoblock-Wärmepumpe (Heisha/Jeisha) mittlerweile fast auswendig. Heizungssysteme anderer Hersteller und alle möglichen Varianten von Buderus sind mir eher unbekannt.

Grüße,
Daniel

**Reply by AlexKusnezov** [reply]
vielen Dank für die Antwort. Verwendest du die beiden von dir genannten Sachen in einem Haushalt? Ergänzen die sich irgendwie?

**Reply by the78mole** [reply]
Hi Alex,
also bei mir läuft eine Panasonic "Jeisha" 5 kW Monoblock-Wärmepumpe (mit [Heishamon](https://www.tindie.com/products/thehognl/heishamon-communication-pcb/) als Einbindung ins Smart Home) und ein 21 kW Logano Heizkessel mit der Buderus Logamatic 2107 M als Steuerung. Dabei übernimmt die Wärmepumpe etwa 80% der Heizwärme und Öl kommt nur dazu, wenn es nicht reicht oder der Strom gerade sehr teuer ist. Manchmal darf die Ölheizung auch das Warmwasser machen :-P
Ergänzen... Naja, die Ölheizung war halt schon da und ich wollte die Wärmepumpe nicht massiv überdimensionieren. Sobald ich das Haus fertig saniert habe, wird die Wärmepumpe das alleine schaffen und die Ölheizung könnte rausfliegen. Das rechnet sich aber nur deshalb, weil ich die komplette Heizungsinstallation selbst mache. Hätte das ein Sanitärunternehmen gemacht, wäre es deutlich wirtschaftlicher, die Ölheizung von Anfang an rauszuwerfen und das Haus vorher zu sanieren.
Grüße,
Daniel

**Reply by AlexKusnezov** [reply]
ok haja klar verstehe. Bei mir ist schon seit knapp zwei Jahren Holz die Hauptwärmequelle, da ich das günstig bekomme und größtenteils im Home Office bin um regelmäßig zu befüllen. Damit kann ich dann alles machen (HK1, HK2 und Warmwasser).

**Comment by the78mole** [comment]
Naja, Holz bekomme ich auch von meinem Paps geliefert, aber damit kann ich nur das Wohnzimmer heizen. Evtl. werde ich mal einen Ofen mit Wasserführung einbauen. Im Technikraum ist alles dafür vorbereitet. Auf der anderen Seite ist Holz halt ziemlich dreckig. Was da an Feinstaub durch den Kamin geht ist auch ziemlich heftig, selbst wenn man einen guten Ofen hat, trockes Holz hat und entsprechend schürt. Dagegen ist jeder Diesel-PKW eine Frischluftquelle.
Außerdem kostet mich die Wärme je nach Strompreis (Tibber) und COP/Außentemperatur effektiv zwischen 1,5 und 12 ct/kWh, im Mittel dürften es so gegen 6 ct/kWh sein. In meinen Jugendjahren habe ich wahrscheinlich so um die 1000 Steer Holz mit meinem Vater zusammen gemacht, und wenn ich das ins Verhältnis mit den 6 ct/kWh setze, dann ist es mir das eigentlich nicht wert 😝

**Reply by AlexKusnezov** [reply]
Tibber scheint eine geile Sache zu sein. Darauf denke ich auch die ganze Zeit rum. Welche Aufwendungen hattest du, um das bei die einzurichten?

**Reply by the78mole** [reply]
Was genau meinst Du denn mit Aufwendungen? Der Tibber-Pulse kostet für Tibber-Neukunden glaube momentan 100€ (wenn man es schlau macht, dann kauft man den mit der 50€ Prämie und bekommt ihn somit für 50€). Mein Smart Home (Home Assistant) war schon da und in wenigen Minuten mit der Tibber-API verbunden. Die ganzen Regeln muss man dann natürlich noch erstellen (wenn man nicht die Tibber-App zum steuern der unterstützten Power-Ups verwendet). Das nimmt dann schon etwas mehr Zeit (einige Stunden) in Anspruch, wenn man viele verschiedene Verbaucher (Wärmepumpe, Batteriespeicher, WallBox, Entfeuchter,...) preisabhängig steuern will... Viele Geräte kann man natürlich Out-of-the-Box integrieren (z.B. die go-e Gemini Charger Wallbox), für andere braucht man dann eine schaltbare Steckdose oder muss die fremde API anbinden (z.B. Victron-Batteriespeicher).
Werde da gerne auch etwas ausführlicher, wenn Du mir sagst, was genau Du wissen möchtest 😉
